diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/BooleanExpressionsRule.kt

```diff
@@ -0,0 +1,165 @@
+        }
+            ?: RuleSet.simplify(expr)
+        if (expr != simplifiedExpression) {
+            COMPLEX_BOOLEAN_EXPRESSION.warnAndFix(configRules, emitWarn, isFixMode, node.text, node.startOffset, node) {
+                fixBooleanExpression(node, simplifiedExpression, mapOfExpressionToChar)
+            }
+        }
+    }
+
+    private fun makeCorrectExpressionString(node: ASTNode, mapOfExpressionToChar: HashMap<String, Char>): String {
+        // `A` character in ASCII
+        var characterAsciiCode = 'A'.code
+        node
+            .findAllNodesWithCondition({ it.elementType == BINARY_EXPRESSION })
+            .filterNot { it.text.contains("&&") || it.text.contains("||") }
+            .forEach {
+                if (mapOfExpressionToChar.containsKey(it.text)) {
+                    return@forEach
+                }
+                // Assigning the boolean expression to a char. The lib represents expressions[a > 5] as chars [A, B, C ...].
+                mapOfExpressionToChar[it.text] = characterAsciiCode.toChar()
+                characterAsciiCode++
+            }
+        // Library is using & as && and | as ||.
+        var correctedExpression = "(${node
+            .text
+            .replace("&&", "&")
+            .replace("||", "|")})"
+        mapOfExpressionToChar.forEach { (refExpr, char) ->
+            correctedExpression = correctedExpression.replace(refExpr, char.toString())
+        }
+        return correctedExpression
+    }
+
+    private fun fixBooleanExpression(
+        node: ASTNode,
+        simplifiedExpr: Expression<String>,
+        mapOfExpressionToChar: HashMap<String, Char>) {
+        var correctKotlinBooleanExpression = simplifiedExpr
+            .toString()
+            .replace("&", "&&")
+            .replace("|", "||")
+            .drop(1)  // dropping first (
+            .dropLast(1)  // dropping last )
+        mapOfExpressionToChar.forEach { (key, value) ->
+            correctKotlinBooleanExpression = correctKotlinBooleanExpression.replace(value.toString(), key)
+        }
+        node.replaceChild(node.firstChildNode, KotlinParser().createNode(correctKotlinBooleanExpression))
+    }
+
+    /**
+     * Checks if boolean expression can be simplified with distributive law.
+     *
+     * @return String? null if it cannot be simplified. Simplified string otherwise.
+     */
+    private fun checkDistributiveLaw(
+        expr: Expression<String>,
+        mapOfExpressionToChar: HashMap<String, Char>,
+        node: ASTNode): String? {
+        // checking that expression can be considered as distributive law
+        if (!isDistributiveLaw(node, expr.toString(), mapOfExpressionToChar)) {
+            return null
+        }
+        val expressionsLogicalOperator = expr.toString().first { it == '&' || it == '|' }
+        // we return expression depending on second operator
+        return if (expressionsLogicalOperator == '&') {
+            "A & (B | C)"
```

Comment:
How does it work? What if there are more variables in the original expression? Also, why do you need to check distributive law manually? Doesn't jbool do it as well?

Reply:
Jbool doesn't simplify distributive law for some reason.

Reply:
Ok, but what about greater number of variables?

Reply:
Yeah that's a bug. Fixed